evm/chains/base.py

```diff
@@ -609,6 +609,8 @@ def validate_block(self, block: BaseBlock) -> None:
         Since block validation (specifically the uncle validation) must have
         access to the ancestor blocks, this validation must occur at the Chain
         level.
+
+        Cannot be used to validate genesis block.
```

Comment:
Should this be explicitely handled to raise an exception if `block.header.is_genesis`?